internal/users/users_test.go

```diff
@@ -121,17 +122,174 @@ func TestUpdateLocalGroups(t *testing.T) {
 				return
 			}
 
-			got := usertests.IdemnpotentOutputFromGPasswd(t, destCmdsFile)
+			got := usertests.IdempotentGPasswdOutput(t, destCmdsFile)
 			want := testutils.LoadWithUpdateFromGolden(t, got)
 			require.Equal(t, want, got, "UpdateLocalGroups should do the expected gpasswd operation, but did not")
 		})
 	}
 }
 
+func TestCleanupSystemGroups(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		groupFilePath string
+
+		getUsersArgs []string
+
+		wantErr bool
+	}{
+		"No-op when there are no inactive users":        {groupFilePath: "user_in_many_groups.group"},
+		"Cleans up user from group":                     {groupFilePath: "inactive_user_in_one_group.group"},
+		"Cleans up user from multiple groups":           {groupFilePath: "inactive_user_in_many_groups.group"},
+		"Cleans up multiple users from group":           {groupFilePath: "inactive_users_in_one_group.group"},
+		"Cleans up multiple users from multiple groups": {groupFilePath: "inactive_users_in_many_groups.group"},
+
+		"Error when NSS errors out":                   {groupFilePath: "inactive_user_in_one_group.group", getUsersArgs: []string{"nss-fail"}, wantErr: true},
+		"Error on missing groups file":                {groupFilePath: "does_not_exists.group", wantErr: true},
+		"Error when groups file is malformed":         {groupFilePath: "malformed_file.group", wantErr: true},
+		"Error on any unignored delete gpasswd error": {groupFilePath: "gpasswdfail_in_deleted_group.group", wantErr: true},
+	}
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			destCmdsFile := filepath.Join(t.TempDir(), "gpasswd.output")
+			groupFilePath := filepath.Join("testdata", tc.groupFilePath)
+			gpasswdCmd := []string{"env", "GO_WANT_HELPER_PROCESS=1",
+				fmt.Sprintf("GO_WANT_HELPER_PROCESS_DEST=%s", destCmdsFile),
+				fmt.Sprintf("GO_WANT_HELPER_PROCESS_GROUPFILE=%s", groupFilePath),
+				os.Args[0], "-test.run=TestMockgpasswd", "--",
+			}
+
+			getUsersCmd := []string{"env", "GO_WANT_HELPER_PROCESS=1", os.Args[0], "-test.run=TestMockgetusers", "--"}
+			if tc.getUsersArgs == nil {
+				tc.getUsersArgs = []string{"myuser", "otheruser", "otheruser2", "otheruser3", "otheruser4"}
+			}
+			getUsersCmd = append(getUsersCmd, strings.Join(tc.getUsersArgs, ","))
+
+			cleanupOptions := []users.Option{
+				users.WithGpasswdCmd(gpasswdCmd),
+				users.WithGroupPath(groupFilePath),
+				users.WithGetUsersCmd(getUsersCmd),
+			}
+			err := users.CleanSystemGroups(cleanupOptions...)
+			if tc.wantErr {
+				require.Error(t, err, "CleanupSystemGroups should have failed")
+				return
+			}
+			require.NoError(t, err, "CleanupSystemGroups should not have failed")
+
+			// Always check the golden files missing for no-op too on error
```

Comment:
So, you copied that from the other tests. However, the other tests don’t return on error and continue checking. In that tests, you return on any errors…